src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -36,73 +37,106 @@ class SaaSConnector(BaseConnector[AuthenticatedClient]):
 
     def __init__(self, configuration: ConnectionConfig):
         super().__init__(configuration)
-        self.secrets = configuration.secrets
-        self.saas_config = configuration.get_saas_config()
-        self.client_config = self.saas_config.client_config  # type: ignore
-        self.endpoints = self.saas_config.top_level_endpoint_dict  # type: ignore
-        self.collection_name: Optional[str] = None
-        self.privacy_request: Optional[PrivacyRequest] = None
+        required_saas_config = configuration.get_saas_config()
+        assert required_saas_config is not None
+        self.saas_config = required_saas_config
+        self.endpoints = self.saas_config.top_level_endpoint_dict
+        self.secrets = cast(Dict, configuration.secrets)
+        self.current_collection_name: Optional[str] = None
+        self.current_privacy_request: Optional[PrivacyRequest] = None
+        self.current_saas_request: Optional[SaaSRequest] = None
 
     def query_config(self, node: TraversalNode) -> SaaSQueryConfig:
         """
         Returns the query config for a given node which includes the endpoints
         and connector param values for the current collection.
         """
-        # store collection_name for logging purposes
-        self.collection_name = node.address.collection
+        privacy_request = self.current_privacy_request
+        assert privacy_request is not None
         return SaaSQueryConfig(
             node,
             self.endpoints,
-            self.secrets,  # type: ignore
-            self.saas_config.data_protection_request,  # type: ignore
-            self.privacy_request,  # type: ignore
+            self.secrets,
+            self.saas_config.data_protection_request,
+            privacy_request,
         )
 
+    def get_client_config(self) -> ClientConfig:
+        """Utility method for getting client config according to the current class state"""
+        saas_config_client_config = self.saas_config.client_config
+        current_request_client_config = (
+            self.current_saas_request.client_config
+            if self.current_saas_request and self.current_saas_request.client_config
```

Comment:
is it valid for us to get here with `self.current_saas_request` not being set? i'm wondering whether we should make a similar assertion as you make for the `privacy_request` above, to effectively ensure that the connector state has been set as is needed here.

i may be off with this though, so feel free to ignore this suggestion if it doesn't make sense.

Reply:
I had this same thought but wasn't sure where to put it. I'll add the assertion to `get_client_config` and `get_rate_limit_config`. This will work nicely with not having to do null checks.